src/options/imports/index_ops.js

```diff
@@ -0,0 +1,56 @@
+import index from 'src/search/search-index'
+
+const db = index.db._db
+
+async function downloadChunk(data, counter) {
+    chrome.downloads.download({
```

Comment:
Best to use the `browser` global instead of `chrome` as that's the API we're explicitly polyfilling - same API but Promise-based.

 You could also try specifying the [`saveAs`](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/downloads/download) param in the download call to set dialog skip prefs.

Also, a `return` should be needed with the `downloads.download()` call, so that the top-level `downloadChunk` function returns that same Promise which will resolve when the async download is done. Then the L37 (within the `do..while`) call can properly `await` resolution and `catch` any exceptions.

Reply:
I didn't understand this part, `downloads.download()` needs a callback?

Reply:
No, it returns a `Promise` instead, but that should be pushed up the callstack with a `return` so all callers know the status of whatever async logic they are expecting to wait on. Needs `return` on L4 of `index_ops` module, or the `await downloadChunk()` call on L36 isn't guaranteed to wait for the `browser.downloads.download` execution to resolve before calling the next chunk.

Reply:
So I should use a `Promise` constructor like in `grabDBChunk`?

Reply:
Nah because it already returns an instance of `Promise`.